Packages/ClientRuntime/Sources/Middleware/OperationStack.swift

```diff
@@ -7,76 +7,68 @@ public struct OperationStack<OperationStackInput: Encodable & Reflection,
-        self.finalizeStep = FinalizeStep<OperationStackOutput,
-                                         OperationStackError>(id: FinalizeStepId)
-        self.deserializeStep = DeserializeStep<OperationStackOutput,
-                                               OperationStackError>(id: DeserializeStepId)
+                                           OperationStackOutput>(id: SerializeStepId)
+        self.buildStep = BuildStep<OperationStackOutput>(id: BuildStepId)
+        self.finalizeStep = FinalizeStep<OperationStackOutput>(id: FinalizeStepId)
+        self.deserializeStep = DeserializeStep<OperationStackOutput>(id: DeserializeStepId)
         
     }
     
     /// This execute will execute the stack and use your next as the last closure in the chain
     public func handleMiddleware<H: Handler>(context: HttpContext,
                                              input: OperationStackInput,
-                                             next: H) -> SdkResult<OperationStackOutput, OperationStackError>
+                                             next: H) async throws -> OperationStackOutput
     where H.Input == SdkHttpRequest,
           H.Output == OperationOutput<OperationStackOutput>,
-          H.Context == HttpContext,
-          H.MiddlewareError == SdkError<OperationStackError> {
-        
-        let deserialize = compose(next: DeserializeStepHandler(handler: next), with: deserializeStep)
-        let finalize = compose(next: FinalizeStepHandler(handler: deserialize), with: finalizeStep)
-        let build = compose(next: BuildStepHandler(handler: finalize), with: buildStep)
-        let serialize = compose(next: SerializeStepHandler(handler: build), with: serializeStep)
-        let initialize = compose(next: InitializeStepHandler(handler: serialize), with: initializeStep)
-        
-        let result = initialize.handle(context: context, input: input)
-        return result.flatMap { operationOutput in
-            return .success(operationOutput.output!)
-        }
-    }
+          H.Context == HttpContext {
+              
+              let deserialize = compose(next: DeserializeStepHandler(handler: next), with: deserializeStep)
+              let finalize = compose(next: FinalizeStepHandler(handler: deserialize), with: finalizeStep)
+              let build = compose(next: BuildStepHandler(handler: finalize), with: buildStep)
+              let serialize = compose(next: SerializeStepHandler(handler: build), with: serializeStep)
+              let initialize = compose(next: InitializeStepHandler(handler: serialize), with: initializeStep)
+              
+              let result = try await initialize.handle(context: context, input: input)
+              guard let output = result.output else {
+                  throw ClientError.unknownError("Something went terribly wrong where the output was not set on the response. Please open a ticket with us at https://github.com/awslabs/aws-sdk-swift")
```

Comment:
Isn't it possible for requests not to have a response -- e.g. a http 204 response would indicate no body?  i don't think we should assume this is a client error, and seems like we should do some more thinking about how to handle this.

Reply:
I reviewed the protocol codegen tests for this -- and this is not a problem.  The only thing that we may need to update (in a separate PR) is that the generated test in AWSJson1_1/EmptyOperationResponseTest/testhandles_empty_output_shape is not doing any asserts.  We'll need to update our codegen to make sure it is comparing actual and expected.